index.js

```diff
@@ -0,0 +1,155 @@
+'use strict';
+
+const RtmClient = require('@slack/client').RtmClient;
+const RTM_EVENTS = require('@slack/client').RTM_EVENTS;
+const CLIENT_EVENTS = require('@slack/client').CLIENT_EVENTS;
+const schedule = require('node-schedule');
+
+const { util } = require('./app');
+
+
+const botToken = process.env.SLACK_BOT_TOKEN || '';
+const darkskyToken = process.env.DARKSKY_TOKEN || '';
+
+// fixed coordinates used to query darksky
+const dcLocation = {
+    latitude: 38.8892681,
+    longitude: -77.0501425
+};
+
+if (!botToken || !darkskyToken) {
+    console.error('SLACK_BOT_TOKEN and DARKSKY_TOKEN must be passed as environment variables!');
+    process.exit(1);
+}
+
+var rtm = new RtmClient(botToken);
+
+let botID;
+let botChannel;
+
+rtm.on(CLIENT_EVENTS.RTM.AUTHENTICATED, (rtmStartData) => {
+    // XXX the private surry-channel doesn't show up in the rtmStartData even though my
+    // bot was invited to the channel :(
+
+    // looks like self.id should always be present - https://api.slack.com/methods/rtm.start
+    botID = rtmStartData.self.id.toLowerCase();
+    console.log(`Logged in as ${rtmStartData.self.name}(${botID}) of team ${rtmStartData.team.name}, but not yet connected to a channel`);
+});
+
+//let rtmConnectionOpened = false;
+
+function sendMessage(message, channel) {
+    if (rtm.connected) {
+        rtm.sendMessage(message, channel);
+    }
+}
+
+rtm.on(RTM_EVENTS.MESSAGE, function handleRtmMessage(message) {
+
+    if (!message.text) {
+        return;
+    }
+
+    // forgive trailing spaces and ignore case
+    const normalizedMessage = message.text.trim().toLowerCase();
+    let now = false;
+    let tomorrow = false;
+
+    // only respond to messages where we are mentioned
+    if (normalizedMessage.indexOf(`<@${botID}>`) === -1) {
+        return;
+    }
+
+    // private channel ID isn't available on start, so respond on the channel
+    // we're mentioned on.
+    botChannel = message.channel;
+
+    if (normalizedMessage.indexOf('weather now') !== -1) {
```

Comment:
If the command API was made a bit more formal so it's easier to pull out the command + arguments from a message (e.g. command is always at the beginning of the message), I could parse out the command and then use an object that maps command names to functions that handle those commands.

Then I would just need a check to make sure the command maps to a function and I wouldn't need a new `if` for every command.